pkg/validation/options.go

```diff
@@ -235,7 +237,78 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 
 	switch p := o.GetProvider().(type) {
 	case *providers.AzureProvider:
-		p.Configure(o.Providers[0].AzureConfig.Tenant)
+		p.ConfigureTenant(o.Providers[0].AzureConfig.Tenant)
```

Comment:
We are trying to move away from this validation package mutating the providers etc during validations. Any way we can make this validation only and then load the certificates etc when the `NewAzureProvider` is instantiated?